neuralmonkey/decoders/sequence_labeler.py

```diff
@@ -130,7 +130,8 @@ def train_xents(self) -> tf.Tensor:
     def cost(self) -> tf.Tensor:
         # Cross entropy mean over all words in the batch
         # (could also be done as a mean over sentences)
-        return tf.reduce_sum(self.train_xents) / tf.reduce_sum(self.train_mask)
+        return ((tf.reduce_sum(self.train_xents) + 1)
```

Comment:
Proč ty +1. Stává se ti někdy, že chceš labelovat prázdnou sekvenci? Nebylo v tom případě lepší přidat něco jako `1e-9` do jmenovatele, to ho normálně nezmění a když nebudeš mít žádnej target, tak stejně budeš mít nulu v čitateli.

Reply:
U toho skriptu preprocess_bert se mi obcas stavalo, ze to celou sekvenci "zamaskovalo". Pak je samozrejme sance, ze se ti do batche dostanou jen takoveto sekvence a pak skutecne muze vyjit sum pres masky 0.
Tohleto vzdy zajisti, ze tam nebude div. by zero a pri rozumne velkych batchich (> 500 nemaskovanych tokenu), by tenhle smoothing nemel prilis menit vysledek.

Reply:
Stejně bych tam dal radši něco malýho do jmenovatele, abysme se někdy někdo nedivili, proč se ten labelátor chová divně s malou batchí.

Reply:
Ok, klidne tam muzeme dat standardne 1e-9. Kdyz jsem to resil, tak jsem mel pocit, ze v nejakych pripadech by to mohlo delat neplechu.